core/state/snapshot/hextrie_generator.go

```diff
@@ -33,6 +35,46 @@ type leaf struct {
 
 type trieGeneratorFn func(in chan (leaf), out chan (common.Hash))
 
+func GenerateBinaryTree(path string, it AccountIterator) common.Hash {
+	db, err := rawdb.NewLevelDBDatabase(path+"/bintrie", 128, 1024, "")
+	if err != nil {
+		panic(fmt.Sprintf("error opening bintrie db, err=%v", err))
+	}
+	btrie := new(trie.BinaryTrie)
+	btrie.CommitCh = make(chan trie.BinaryHashPreimage)
+
+	var nodeCount uint64
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for kv := range btrie.CommitCh {
+			nodeCount++
+			db.Put(kv.Key, kv.Value)
```

Comment:
Unless you depend on the commit being done synchronously. If you don't require it to be synchronous, you could do two things: 
1. Do batching here, 
2. Add a buffer to the `CommitCh`. Sometimes it will lag behind, but other times the db insert may be fast (when it only puts it into batch), so a reasonably sized buffer may be good here.